### DISCUSSION TITLE
X310 dc spike issues?

### DISCUSSION THREAD

**Discussion by teamm8** [srsran/srsRAN_Project#111] [General Help] [answered]
Hi, 

https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html
I have successfully followed this application note to setup srsran project with a 5G SA and srsUE using both ZMQ and UHD. However, using UHD, the connection between the ue and gnb is unrealiable and the data rate is low. Specifically I am using 2 usrp2954R which have been converted to an x310 for use with uhd 4.1.0.5. They are connected via a couple of splitters and attenuators rather than over the air. 

I am using 1500 MTU because I am running over 1GB connection. Setting to 9000 caused connection issues due to latency. I am setting my gnb sample rate to 11.52MS/s which is under the uhd benchmark_rate which works fine up to ~25MS/s. While the gnb and ue are running, I get one under run at start up other wise the connection between the host and usrp is stable using these settings with no errors. So I am confident the connection to the radio is stable. 

Looking at the RF, there is a dc spike which is over 10-20db above the gnb signal which is only 10db above the noise floor. Reading around uhd, I have tried to uhd_cal procedures and have stumbled upon a method of implemetning an lo offset but I am sturggling to find a way of giving this to uhd via the the config file for srs gnb. 

I'm happy to post logs but I think the fix is to implement a method of manually setting the lo offset but would appreciate the discussion. 

In summary, does srsRAN have a method of shifting the IQ generated away from the centre frequency?

Many thanks for looking!

**Comment by ismagom** [comment, marked as answer]
Hi, there is an option `lo_offset` under `ru_sdr` section. Set it to half your BW and thia should move LO away

**Reply by teamm8** [reply]
Many thanks, worked perfectly in shifting the LO. 
Is there a way to list all settings available for the config files or somewhere to point me to in the source code where they are first parsed to avoid future queries?

**Reply by ismagom** [reply]
Just type - -help

**Comment by mrouili** [comment]
Hi @teamm8, do you mind sharing your gNB configuration file for the x310? I am using the same USRP and I'm facing some issues upon deployment and I can't seem to figure out why. Just trying to narrow down the problem before creating a separate issue thread. Many thanks!

**Reply by teamm8** [reply]
Hi @mrouili 
After a lot more trial and error, I found that the setup was not working and I have had to move to a more cpu intense system. The new hardware setup has allowed me to connect via 10G SFP interfaces and my connectivity problems I was having have gone away. 
Specifically, I was finding that the two systems (srsgnb and srsue on seperate laptops) were happily synchronised if data was not being passed through the system. I was using an external ref clock provided from a spectrum analyser. As soon as I started iperf, a sample would be dropped and the link was not able to recover. The network buffer in linux was fine and keeping below the net.core.r_mem_max figure but every few seconds, the network monitor would report a complete drop in connectivity. This could have been a local issue and therefore not raised as a discussion topic. Apologies, I had to park my work but wanted to get back to you and give you an update
